typescript/agentkit/src/wallet-providers/privyWalletProvider.ts

```diff
@@ -1,44 +1,78 @@
 import { PrivyEvmWalletProvider, PrivyEvmWalletConfig } from "./privyEvmWalletProvider";
 import { PrivySvmWalletProvider, PrivySvmWalletConfig } from "./privySvmWalletProvider";
+import {
+  PrivyEvmEmbeddedWalletProvider,
+  PrivyEvmEmbeddedWalletConfig,
+} from "./privyEvmEmbeddedWalletProvider";
 
-export type PrivyWalletConfig = PrivyEvmWalletConfig | PrivySvmWalletConfig;
+export type PrivyWalletConfig =
+  | PrivyEvmWalletConfig
+  | PrivySvmWalletConfig
+  | PrivyEvmEmbeddedWalletConfig;
+
+export type PrivyWalletProviderVariant<T> = T extends { walletType: "embedded" }
+  ? PrivyEvmEmbeddedWalletProvider
+  : T extends { chainType: "solana" }
+    ? PrivySvmWalletProvider
+    : PrivyEvmWalletProvider;
 
 /**
  * Factory class for creating chain-specific Privy wallet providers
  */
 export class PrivyWalletProvider {
   /**
-   * Creates and configures a new wallet provider instance based on the chain type.
+   * Creates and configures a new wallet provider instance based on the chain type and wallet type.
    *
    * @param config - The configuration options for the Privy wallet
-   * @returns A configured WalletProvider instance for the specified chain
+   * @returns A configured WalletProvider instance for the specified chain and wallet type
    *
    * @example
    * ```typescript
-   * // For EVM (default)
+   *  For EVM server wallets (default)
    * const evmWallet = await PrivyWalletProvider.configureWithWallet({
    *   appId: "your-app-id",
    *   appSecret: "your-app-secret"
    * });
    *
-   * // For Solana
+   * For Solana
    * const solanaWallet = await PrivyWalletProvider.configureWithWallet({
    *   appId: "your-app-id",
    *   appSecret: "your-app-secret",
    *   chainType: "solana"
    * });
+   *
+   * For Embedded Wallet
+   * const embeddedWallet = await PrivyWalletProvider.configureWithWallet({
+   *   appId: "your-app-id",
+   *   appSecret: "your-app-secret",
+   *   walletId: "delegated-wallet-id",
+   *   walletType: "embedded"
+   * });
    * ```
    */
   static async configureWithWallet<T extends PrivyWalletConfig>(
-    config: T & { chainType?: "ethereum" | "solana" },
-  ): Promise<T extends { chainType: "solana" } ? PrivySvmWalletProvider : PrivyEvmWalletProvider> {
+    config: T & {
+      chainType?: "ethereum" | "solana";
+      walletType?: "server" | "embedded";
+    },
+  ): Promise<PrivyWalletProviderVariant<T>> {
+    // Check for embedded wallet first
+    if (config.walletType === "embedded") {
+      return PrivyEvmEmbeddedWalletProvider.configureWithWallet(
+        config as PrivyEvmEmbeddedWalletConfig,
+      ) as unknown as PrivyWalletProviderVariant<T>;
```

Comment:
You don't need these `as unknown`. There are two solutions:

1. That function returns a promise, so just go `as Promise<PrivyWalletProviderVariant<T>`
```ts
return PrivyEvmEmbeddedWalletProvider.configureWithWallet(
  config as PrivyEvmEmbeddedWalletConfig,
) as Promise<PrivyWalletProviderVariant<T>>;
```
3. await the response first, i.e.
```ts
return (await PrivyEvmEmbeddedWalletProvider.configureWithWallet(
  config as PrivyEvmEmbeddedWalletConfig,
)) as PrivyWalletProviderVariant<T>;
```